File: app/src/main/java/br/com/popularmovies/services/movieService/source/local/MovieDao.java
package br.com.popularmovies.services.movieService.source.local;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import java.util.List;

import br.com.popularmovies.services.movieService.response.Movie;

@Dao
public interface MovieDao {

    @Query("SELECT * FROM movie")
    LiveData<List<Movie>> getMovies();

    @Query("SELECT * FROM movie WHERE id = :movieId")
    LiveData<Movie> getMovie(int movieId);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAllMovies(List<Movie> movies);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertMovie(Movie movie);

    @Delete
    void deleteMovie(Movie movie);

    @Query("UPDATE movie SET isFavorite = :status WHERE id = :movieId")
    void saveFavorites(int movieId, boolean status);
}

File: app/src/main/java/br/com/popularmovies/services/movieService/source/local/MovieDao.kt
package br.com.popularmovies.services.movieService.source.local

import androidx.room.*
import br.com.popularmovies.services.movieService.response.MovieDto

@Dao
interface MovieDao {

    @Query("SELECT * FROM movie")
    suspend fun movies(): List<MovieDto>

    @Query("SELECT * FROM movie WHERE id = :movieId")
    suspend fun getMovie(movieId: Int): MovieDto

    @Query("SELECT * FROM movie where isFavorite=:isFavorite")
    suspend fun getFavoriteMovies(isFavorite: Boolean): List<MovieDto>

    @Query("SELECT EXISTS(SELECT * FROM movie WHERE id = :movieId)")
    suspend fun isMovieExists(movieId: Int): Boolean

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertAllMovies(movieDtos: List<MovieDto>)

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun insertMovie(movieDto: MovieDto)

    @Delete
    suspend fun deleteMovie(movieDto: MovieDto)

    @Query("UPDATE movie SET isFavorite = :status WHERE id = :movieId")
    suspend fun saveFavorites(movieId: Int, status: Boolean)
}
